src/keymap_manager.hh

```diff
@@ -43,6 +43,10 @@ public:
     };
     const KeymapInfo& get_mapping(Key key, KeymapMode mode) const;
 
+    using UserModeList = Vector<String>;
+    const UserModeList user_modes() const { return m_user_modes; }
+    void add_user_mode(const String& user_mode_name);
```

Comment:
We do not pass const ref to strings in Kakoune code anymore (except very special cases), we either use a StringView (passed by value) or, if we are going to copy the string, a String by value that gets std::move'd. In the current case, both seems acceptable to me.